ethers-providers/src/provider.rs

```diff
@@ -1161,6 +1161,278 @@ impl<P: JsonRpcClient> Middleware for Provider<P> {
     }
 }
 
+use crate::PinBoxFut;
+use serde::ser::SerializeTuple;
+use std::{
+    collections::HashMap,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+impl<P: JsonRpcClient> Provider<P> {
+    pub fn call_builder<'a>(&'a self, tx: &'a TypedTransaction) -> Caller<'a, P> {
```

Comment:
needs docs and perhaps an example

I think we should move this to the existing `impl Provider` section